### PR TITLE
Send User-Agent info with HTTP requests

### PR DESCRIPTION
**Pull request by michaelnoonan**
Relates to https://github.com/OctopusDeploy/Issues/issues/3359

### CODE REVIEW

File: source/Octopus.Client/OctopusClient.cs [hunk 1]
```diff
@@ -446,6 +447,9 @@ protected virtual OctopusResponse<TResponseResource> DispatchRequest<TResponseRe
             webRequest.Credentials = serverEndpoint.Credentials ?? CredentialCache.DefaultNetworkCredentials;
             webRequest.Method = request.Method;
             webRequest.Headers[ApiConstants.ApiKeyHttpHeaderName] = serverEndpoint.ApiKey;
+            var userAgent = $"OctopusClient-dotnet/{GetType().GetSemanticVersion().ToNormalizedString()}";
+            webRequest.UserAgent = userAgent;
+            webRequest.Headers[ApiConstants.OctopusUserAgentHeaderName] = userAgent;
```
Comment (matt-richardson) [inline source/Octopus.Client/OctopusClient.cs]: Needs a comment to say why we are setting both

File: source/Octopus.Client/OctopusAsyncClient.cs [hunk 2]
```diff
@@ -70,6 +71,9 @@ protected OctopusAsyncClient(OctopusServerEndpoint serverEndpoint, OctopusClient
             client.Timeout = options.Timeout;
             client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
             client.DefaultRequestHeaders.Add(ApiConstants.ApiKeyHttpHeaderName, serverEndpoint.ApiKey);
+            var userAgent = $"OctopusClient-dotnet/{GetType().GetSemanticVersion().ToNormalizedString()}";
```
Comment (matt-richardson) [inline source/Octopus.Client/OctopusAsyncClient.cs]: Is this reflection gonna be slow? worth caching?

File: source/Octopus.Client.Tests/PublicSurfaceAreaFixture.ThePublicSurfaceAreaShouldNotRegress..NETFramework.approved.txt [hunk 3]
```diff
@@ -911,6 +911,12 @@ Octopus.Client.Exceptions
 }
 Octopus.Client.Extensions
 {
+  abstract class AssemblyExtensions
```
Comment (matt-richardson) [inline source/Octopus.Client.Tests/PublicSurfaceAreaFixture.ThePublicSurfaceAreaShouldNotRegress..NETFramework.approved.txt]: this shouldn't be public?

Comment (michaelnoonan) [inline source/Octopus.Client.Tests/PublicSurfaceAreaFixture.ThePublicSurfaceAreaShouldNotRegress..NETFramework.approved.txt]: It's an extension method(s) class - I've made it internal

File: source/Octopus.Client/OctopusAsyncClient.cs (hunk 2 above)
Comment (michaelnoonan) [inline source/Octopus.Client/OctopusAsyncClient.cs]: Now caching into a field on ctor

File: source/Octopus.Client/OctopusClient.cs (hunk 1 above)
Comment (michaelnoonan) [inline source/Octopus.Client/OctopusClient.cs]: Done

**Comment by zentron** [comment]
Looks good. 👍

**Comment by matt-richardson** [comment]
LGTM

**Pull request merged**